### ISSUE TITLE
MAXQUOTAS refined warnings

### ISSUE THREAD

**Issue by rdmark** [Netatalk/netatalk#1101]
After the recent fixes to quota, I'm seeing these warnings when building with meson on Debian 12:

```
dmark@alien:~/dev/netatalk$ ninja -C build
ninja: Entering directory `build'
[165/267] Compiling C object etc/afpd/libafpd.a.p/directory.c.o
In file included from ../etc/afpd/directory.c:45:
../etc/afpd/unix.h:85: warning: "MAXQUOTAS" redefined
   85 | #define MAXQUOTAS 2
      | 
In file included from ../etc/afpd/unix.h:52:
/usr/include/linux/quota.h:40: note: this is the location of the previous definition
   40 | #define MAXQUOTAS 3
      | 
[169/267] Compiling C object etc/afpd/libafpd.a.p/filedir.c.o
In file included from ../etc/afpd/filedir.c:37:
../etc/afpd/unix.h:85: warning: "MAXQUOTAS" redefined
   85 | #define MAXQUOTAS 2
      | 
In file included from ../etc/afpd/unix.h:52:
/usr/include/linux/quota.h:40: note: this is the location of the previous definition
   40 | #define MAXQUOTAS 3
      | 
[175/267] Compiling C object etc/afpd/libafpd.a.p/file.c.o
In file included from ../etc/afpd/file.c:40:
../etc/afpd/unix.h:85: warning: "MAXQUOTAS" redefined
   85 | #define MAXQUOTAS 2
      | 
In file included from ../etc/afpd/unix.h:52:
/usr/include/linux/quota.h:40: note: this is the location of the previous definition
   40 | #define MAXQUOTAS 3
      | 
[176/267] Compiling C object etc/afpd/libafpd.a.p/unix.c.o
In file included from ../etc/afpd/unix.c:33:
../etc/afpd/unix.h:85: warning: "MAXQUOTAS" redefined
   85 | #define MAXQUOTAS 2
      | 
In file included from ../etc/afpd/unix.h:52:
/usr/include/linux/quota.h:40: note: this is the location of the previous definition
   40 | #define MAXQUOTAS 3
      | 
[179/267] Compiling C object etc/afpd/libafpd.a.p/volume.c.o
In file included from ../etc/afpd/volume.c:61:
../etc/afpd/unix.h:85: warning: "MAXQUOTAS" redefined
   85 | #define MAXQUOTAS 2
      | 
In file included from ../etc/afpd/unix.h:52:
/usr/include/linux/quota.h:40: note: this is the location of the previous definition
   40 | #define MAXQUOTAS 3
      | 
[181/267] Compiling C object etc/afpd/libafpd.a.p/afpstats_obj.c.o
In file included from /usr/include/glib-2.0/glib/gthread.h:34,
                 from /usr/include/glib-2.0/glib/gasyncqueue.h:34,
                 from /usr/include/glib-2.0/glib.h:34,
                 from /usr/include/glib-2.0/gobject/gbinding.h:30,
                 from /usr/include/glib-2.0/glib-object.h:24,
                 from ../etc/afpd/afpstats_obj.c:22:
../etc/afpd/afpstats_obj.c: In function 'afpstats_obj_get_type':
/usr/include/glib-2.0/glib/gatomic.h:131:5: warning: argument 2 of '__atomic_load' discards 'volatile' qualifier [-Wincompatible-pointer-types]
  131 |     __atomic_load (gapg_temp_atomic, &gapg_temp_newval, __ATOMIC_SEQ_CST); \
      |     ^~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gthread.h:262:7: note: in expansion of macro 'g_atomic_pointer_get'
  262 |     (!g_atomic_pointer_get (location) &&                             \
      |       ^~~~~~~~~~~~~~~~~~~~
../etc/afpd/afpstats_obj.c:61:13: note: in expansion of macro 'g_once_init_enter'
   61 |         if (g_once_init_enter(&g_define_type_id__volatile)) {
      |             ^~~~~~~~~~~~~~~~~
[183/267] Compiling C object etc/afpd/libafpd.a.p/nfsquota.c.o
In file included from ../etc/afpd/nfsquota.c:62:
../etc/afpd/unix.h:85: warning: "MAXQUOTAS" redefined
   85 | #define MAXQUOTAS 2
      | 
In file included from ../etc/afpd/unix.h:52:
/usr/include/linux/quota.h:40: note: this is the location of the previous definition
   40 | #define MAXQUOTAS 3
      | 
[186/267] Compiling C object etc/afpd/libafpd.a.p/acls.c.o
In file included from ../etc/afpd/acls.c:59:
../etc/afpd/unix.h:85: warning: "MAXQUOTAS" redefined
   85 | #define MAXQUOTAS 2
      | 
In file included from ../etc/afpd/unix.h:52:
/usr/include/linux/quota.h:40: note: this is the location of the previous definition
   40 | #define MAXQUOTAS 3
      | 
[187/267] Compiling C object etc/afpd/libafpd.a.p/quota.c.o
In file included from ../etc/afpd/quota.c:36:
../etc/afpd/unix.h:85: warning: "MAXQUOTAS" redefined
   85 | #define MAXQUOTAS 2
      | 
In file included from ../etc/afpd/unix.h:52:
/usr/include/linux/quota.h:40: note: this is the location of the previous definition
   40 | #define MAXQUOTAS 3
      | 
[267/267] Linking target etc/uams/uams_gss.so
```

**Comment by dgsga** [comment]
@rdmark, fixing...

**Comment by dgsga** [comment]
Closed by PR #1099

**Comment by rdmark** [comment]
Fixing for 2.4 in https://github.com/Netatalk/netatalk/pull/1120
